File: src/backend/facilitymanagement/src/main/java/org/taonaw/facilitymanagement/domain/model/equipment/EquipmentName.java
package org.taonaw.facilitymanagement.domain.model.equipment;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NonNull;
import org.taonaw.facilitymanagement.domain.shared.Assertion;

@Getter
@EqualsAndHashCode
public class EquipmentName {
    private final String value;

    public EquipmentName(@NonNull String value) {
        Assertion.argumentRange(value, 1, 100);
        this.value = value;
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/equipment/EquipmentName.kt
package org.taonaw.facility.domain.model.equipment

data class EquipmentName(val value: String) {

    companion object {
        const val LENGTH_MIN = 1
        const val LENGTH_MAX = 50
    }

    init {
        require(value.length in LENGTH_MIN..LENGTH_MAX) {
            "機材名は $LENGTH_MIN 文字以上 $LENGTH_MAX 文字以下でなければなりません。"
        }
    }
}